### PR TITLE
refactor: Small refactor of reference module and tests

### PR DESCRIPTION
**Pull request by nima**
This is the nice-to-have changes that was split out of #225.

Signed-off-by: Nima Talebi <nima@users.noreply.github.com>

### CODE REVIEW

File: registry/reference.go [hunk 1]
```diff
@@ -56,33 +56,65 @@ type Reference struct {
...
 
-// ParseReference parses a string into a artifact reference.
-// If the reference contains both the tag and the digest, the tag will be
-// dropped.
-// Digest is recognized only if the corresponding algorithm is available.
-func ParseReference(raw string) (Reference, error) {
-	parts := strings.SplitN(raw, "/", 2)
+// ParseReference parses a string (artifact) into an `artifact reference`.
+//
+// Note: An "image" is an "artifact", however, an "artifact" is not necessarily
```
Comment (Wwwsylvia) [inline registry/reference.go:61]: It looks like this sentence is incomplete.

Comment (shizhMSFT) [inline registry/reference.go:61]: It seems this sentence is not complete.

File: registry/reference.go [hunk 2]
```diff
@@ -56,33 +56,65 @@ type Reference struct {
 	Reference string
 }
 
-// ParseReference parses a string into a artifact reference.
-// If the reference contains both the tag and the digest, the tag will be
-// dropped.
-// Digest is recognized only if the corresponding algorithm is available.
-func ParseReference(raw string) (Reference, error) {
-	parts := strings.SplitN(raw, "/", 2)
+// ParseReference parses a string (artifact) into an `artifact reference`.
```
Comment (shizhMSFT) [inline registry/reference.go:59]: image

The doc looks nice.

**Review by shizhMSFT — CHANGES_REQUESTED** [review]

**Review by shizhMSFT — COMMENTED** [review]
It would be even better if we have a [testable example](https://go.dev/blog/examples) for `ParseReference()`

**Comment by shizhMSFT** [comment]
> It would be even better if we have a [testable example](https://go.dev/blog/examples) for `ParseReference()`

We can do this in another PR.

File: registry/reference.go (hunk 1 above)
Comment (nima) [reply registry/reference.go:61]: 🤣 Fixed.

Comment (nima) [reply registry/reference.go:61]: Fixed 😄

**Review by Wwwsylvia — APPROVED** [review]
LGTM

**Comment by Wwwsylvia** [comment]
@nima Can you fix the DCO?

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**